lib/ansible/cli/doc.py

```diff
@@ -71,7 +73,197 @@ class PluginNotFound(Exception):
     pass
 
 
-class DocCLI(CLI):
+class RoleMixin(object):
+    """A mixin containing all methods relevant to role argument specification functionality.
+
+    Note: The methods for actual display of role data are not present here.
+    """
+
+    ROLE_ARGSPEC_FILE = 'argument_specs.yml'
```

Comment:
I want to say we should support `.yaml` extension as well as `.yml`. I thought we were more flexible about this in the `galaxy` code, but it seems we only support `.yml` in many places.

I would consider that extra credit to support `.yaml` but it would be nice.

Reply:
I seem to recall someone suggesting supporting only `.yml` but not sure if I'm imagining that. Will look back through my notes... but yeah, we could easily support that.

Reply:
Yes, choosing only one was suggested by @nitzmahone in the spec:

```
Might want to just pick one for new features, eg, collection runtime.yml only supports .yml extension
```